### PR TITLE
Develop branch

### PR DESCRIPTION
**Pull request by jgriffiths**
Updates for singlesig/backend consistency for testing. The HW device change relates to unpublished test changes that IMO demonstrate the old rename-then-update code was confusing and unintuitive vs naming the device info after the structure in the hw_device gdk parameter which it updates.

The shared tx enablement code applies only to the current gdk development branch but can be merged to master now as (a) its harmless, (b) it enables singlesig cli testing and (c) singlesig is not yet an officially supported cli client.

### CODE REVIEW

File: green_cli/common.py [hunk 1]
```diff
@@ -485,39 +486,43 @@ def sendtransaction(session, details):
...
+    return get_txhash_with_sync(session, details, wait)
 
-def _send_transaction(session, details):
+def _send_transaction(session, details, wait):
     add_utxos_to_transaction(session, details)
     details = gdk_resolve(gdk.create_transaction(session.session_obj, json.dumps(details)))
     details = gdk_resolve(gdk.sign_transaction(session.session_obj, json.dumps(details)))
     details = gdk_resolve(gdk.send_transaction(session.session_obj, json.dumps(details)))
+    return get_txhash_with_sync(session, details, wait)
     return details['txhash']
```
Comment (jkauffman1) [inline green_cli/common.py:498]: Remove this line?

File: green_cli/utils.py [hunk 2]
```diff
@@ -25,3 +25,10 @@ def add_utxos_to_transaction(session, details):
         utxos = gdk_resolve(gdk.get_unspent_outputs(session.session_obj, json.dumps(utxo_details)))
         details['utxos'] = utxos['unspent_outputs']
 
+def get_txhash_with_sync(session, details, wait):
+    txhash = details['txhash']
+    while wait:
```
Comment (jkauffman1) [inline green_cli/utils.py:30]: Do we want some kind of timeout mechanism?

Comment (jgriffiths) [reply green_cli/utils.py:30]: I guess we probably should, I'll add one ta

Comment (jgriffiths) [reply green_cli/utils.py:30]: Added `--timeout` everywhere cheers

**Comment by LeoComandini** [comment]
utACK 53fef8a88e46a33d7a949ad2b27b71a8978cccf1

**Pull request merged**